src/wrapper/jnienv.rs

```diff
@@ -1845,6 +1845,52 @@ impl<'a> JNIEnv<'a> {
         jni_void_call!(self.internal, EnsureLocalCapacity, capacity);
         Ok(())
     }
+
+    /// Bind function pointers to native methods of class
+    /// according to method name and signature
+    /// for details see [documentation](https://docs.oracle.com/javase/8/docs/technotes/guides/jni/spec/functions.html#RegisterNatives)
+    pub fn register_native_methods<'c, T>(&self, class: T, methods: Vec<NativeMethod>) -> Result<()>
+    where
+        T: Desc<'a, JClass<'c>>,
+    {
+        let class = class.lookup(self)?;
+        let jni_native_methods: Vec<JNINativeMethod> = methods
+            .iter()
+            .map(|nm| JNINativeMethod {
+                name: nm.name.as_ptr() as *mut c_char,
+                signature: nm.sig.as_ptr() as *mut c_char,
+                fnPtr: nm.fn_ptr,
+            })
+            .collect();
+        let res = jni_non_void_call!(
+            self.internal,
+            RegisterNatives,
+            class.into_inner(),
+            jni_native_methods.as_ptr(),
+            jni_native_methods.len() as jint
+        );
+        jni_error_code_to_result(res)
+    }
+
+    /// Unbind all native methods of class
+    pub fn unregister_native_methods<'c, T>(&self, class: T) -> Result<()>
+    where
+        T: Desc<'a, JClass<'c>>,
+    {
+        let class = class.lookup(self)?;
+        let res = jni_non_void_call!(self.internal, UnregisterNatives, class.into_inner());
+        jni_error_code_to_result(res)
+    }
+}
+
+/// Native method descriptor
+pub struct NativeMethod {
+    /// Name of method
+    pub name: JNIString,
+    /// Method signature
+    pub sig: JNIString,
```

Comment:
nit: `signature`. We're not limited in chars, are we? :wink:

Reply:
I follow the naming convention taken in this file. In other public methods we use sig as signature name.